### PR TITLE
replication mode: add new intermediate state

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

- add intermediate state `SYNC_WAIT`
- add `StoreDRAutoSyncStatus` to allow tikv to report the global state to PD
- add `available_stores` for PD to indicate the list of stores that TiKV can send raft messages to

### CODE REVIEW

**Comment by disksing** [comment]
Please take a look. @lhy1024 @nolouch

File: proto/replication_modepb.proto [hunk 1]
```diff
@@ -17,10 +17,12 @@ message ReplicationStatus {
 enum DRAutoSyncState {
     // Raft logs need to sync between different DCs
     SYNC = 0;
+    // Wait for switching to ASYNC. Stop sync raft logs between DCs.
+    ASYNC_WAIT = 1;
```
Comment (lhy1024) [inline proto/replication_modepb.proto:21]: why insert rather than append?

**Review by nolouch — APPROVED** [review]
LGTM. currently this feature only used in POC, we can do some incompatible modifications.

**Comment by nolouch** [comment]
cc @lhy1024

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**